0000-account-name-system.md

```diff
@@ -0,0 +1,114 @@
+- Proposal Name: `account_name_system`
+- Start Date: 2019-07-21
+- NEP PR: [nearprotocol/neps#0000](https://github.com/nearprotocol/neps/pull/0000)
+
+# Summary
+[summary]: #summary
+
+The Account Name System is a hierarchical and decentralized naming system for accounts, contracts and applications running on the NEAR blockchain.
+It describes rules with which a new account names can be registered.
+
+# Motivation
+[motivation]: #motivation
+
+NEAR uses readable account names for easier access to accounts and applications. 
+On the other hand this creates a problem of squatting names as well as possible issues with known names be taken by fraudulent entities.
+
+The ANS follows ideas in DNS, where top authority only needs to decide on registering top level domains and the rest is done by domain registrars.
+In this document we will describe rules for top level account names (TLAs) and enumerate few possible designs for sub-registrars that can run on TLAs.
+
+Result of this design should be implemented on the runtime level of the NEAR protocol for TLA registration.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+`CreateAccount` transaction type should contain next fields:
+```protobuf
+message CreateAccountTransaction {
+    /// Nonce of the given transaction for originator's access key.
+    uint64 nonce = 1;
+    /// Originator's account id.
+    string originator_id = 2;
+    /// New account's id.
+    string new_account_id = 3;
+    /// Amount of money to allocate on the new account.
+    Uint128 amount = 4;
+    /// Code to deploy at the given account.
+    bytes code = 5;
+    /// Access keys to attach to given account
+    repeated AccessKey access_keys = 6;
+}
+```
+
+Here, important to note that `CreateAccount` is created by some other account. 
+
+The next rules apply for relationship between `originator_id` and `new_account_id`:
+
+- `new_account_id` can only contain characters from the [a-z0-9@._-] set. And be 5-32 characters long.
```

Comment:
We need to handle shorted TLAs, so this needs to be restricted. E.g. TLA can be 2+ chars. While non TLA should be limited to 5+, but it's not necessary.